src/explore-education-statistics-common/src/components/form/rhf/hooks/useRegister.ts

```diff
@@ -16,7 +16,14 @@ export default function useRegister<
 >(
   name: TFieldName,
   register: UseFormRegister<TFieldValues>,
+  trimInput = false,
 ): UseFormRegisterReturn<TFieldName> {
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  return useMemo(() => register(name), [name]);
+  return useMemo(
+    () =>
+      register(name, {
+        setValueAs: value => (trimInput ? value.trim() : value),
```

Comment:
This is unsafe and could throw an error for non-string values. We'd want to check that the value is actually a string before we try and trim it.

However, we don't actually need this if we just do the trimming in the `FormTextArea` and `FormTextInput` (as mentioned in other places).